### ISSUE TITLE
(storage) getUrl should support contentDisposition (again; used to)

### ISSUE THREAD

**Issue by mitchh** [aws-amplify/amplify-js#13326]
### Is this related to a new or existing framework?

Angular, React, Vue, Next.js

### Is this related to a new or existing API?

Storage

### Describe the feature you'd like to request

`Storage.get` used to allow specification of `Content-Disposition` in the returned URL; in v6 it is no longer possible, with the rationale `Content options (contentDisposition, contentLanguage, contentEncoding, and contentType) are no longer supported in the get API as these values are already provided in the uploaded file.`  While this does apply to most of the other `Content-*` headers, `Content-disposition` is *not* about the content of the file itself, and has other uses. Specifically, consider [this comment from the Android SDK](https://github.com/aws-amplify/aws-sdk-android/blob/adec7ed1ab2debb72449207ad0f6a3d995984211/aws-android-sdk-s3/src/main/java/com/amazonaws/services/s3/model/ResponseHeaderOverrides.java#L27-L30):

```
 * For example, a client could dynamically change the apparent
 * Content-Disposition header of a single object, so that it appears to have a
 * different file name for different callers. One client could be configured
 * return the object with
 *
 * <pre>
 * Content-Disposition: attachment; filename=FileName1.exe
 * </pre>
 *
 * while another could return that same object with headers
 *
 * <pre>
 * Content-Disposition: attachment; filename=FileName2.pdf
 * </pre>
 * ```
 

### Describe the solution you'd like

Add `contentDisposition` (or maybe better yet, a `headers` object that can include `content-disposition`) to `options` in `getUrl`

### Describe alternatives you've considered

Don't have a good alternative ATM

### Additional context

_No response_

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by cwomack** [comment]
Hello, @mitchh 👋.  It indeed is a change from v5 behavior after upgrading to v6 that the `contentDisposition` (as you noted in the docs [here](https://docs.amplify.aws/react/build-a-backend/storage/storage-v5-to-v6-migration-guide/#storageget:~:text=mentioned%20above%3A-,Content%20options,-()).  I'll review this with our team internally, but could you give some further context and use case for this to ensure we understand a broader range of how the `contentDisposition` option will be used?

**Comment by mitchh** [comment]
I have files stored (using Amplify, FWIW) by a previous version of our app
with various `Content-Disposition` header metadata. The most interesting
ones are PDFs. At times I want them to display inline in the browser, and I
also want to provide download URLs.  It seems wasteful to have to manage
multiple copies with different metadata.  That is an S3 design decision
rather than an Amplify issue, but in v5, and in the S3 API underlying it,
the ability to override them in a presigned URL provides a solution.

To be honest, I find the whole notion of storing headers (specifically this
one directing the handling rather than the content) with files a little
odd, though I can see why it's handy much of the time. That said, being
able to override them provides flexibility, and I would rather stay in
Amplify than have to go craft my own URLs.

**Comment by philipws** [comment]
Microsoft Edge default settings breaks S3 presigned URL downloads of document files in my project. Edge attempts to open the files in a web viewer as it believes the link is public, and instead the user gets a 404 (from a Microsoft URL) and assume the AWS based app is broken. 

This is because the setting "open office files in the browser" for downloads is enabled by default. 

https://textslashplain.com/2020/02/26/can-i-in-the-new-edge/ > section "How does Edge render Office documents directly in the browser?"

> The OpenInOfficeViewerIfApplicable feature watches navigations to detect when Office-related file extensions (.xls, .docx, pptx, etc) or Content-Types (application/msword, application/vnd.ms-excel, etc) are observed, indicating that a navigation led directly to an Office document that would otherwise be deemed as having a non-webby MIME type and thus normally be converted into a download.
> 
> The feature is bypassed if:
> 
>   1. The Open Office files in the browser checkbox is turned off in edge://settings
>   2. The response contains a Content-Disposition: attachment header
>   3. The request contains an Authorization or Proxy-Authorization header.
>   4. The user initiated the download from the “Save As” item on the browser’s context menu
>   5. The referring URL is on a limited set of exempted domains (Office, SharePoint, OneDrive, Blackboard, etc)
>   6. The request method is not GET
>   7. The file size is not known (e.g. Transfer-Encoding: chunked)
>   8. The file size is over 100mb
>   9. The browser is running in Incognito mode
>   10. The file was served from an IP address that is not publicly routable

**Comment by cwomack** [comment]
@mitchh, wanted to follow up on this and let you know that this feature is now available on v6.5.0 of Amplify! Can you please upgrade to the latest version and validate this on your side that it's working properly? Thanks!

**Comment by cwomack** [comment]
Closing this issue now that the feature has been released.
